src/pages/admin_groups.js

```diff
@@ -0,0 +1,124 @@
+var shortid = require('shortid')
+
+function list(templates, admin_group, req, res) {
+  if (!req.user || !req.user.isAdmin) {
+    res.sendStatus(403)
+    return
+  }
+
+  admin_group.getAdminGroups().then(admin_groups => {
+    var html = templates.admin_group.list({
+      user: req.user,
+      admin_groups: admin_groups
+    })
+
+    res.send(html)
+  }).catch(err => {
+    console.error(err)
+    res.sendStatus(500)
+  })
+}
+
+function create(templates, admin_group, req, res) {
+  if (!req.user || !req.user.isAdmin) {
+    res.sendStatus(403)
+    return
+  }
+  admin_group.getAdminGroupNames().then((admin_groups) => {
+    var html = templates.admin_group.edit({
+      user: req.user,
+      verb: 'Create',
+      admin_groups: admin_groups
+    })
+
+    res.send(html)
+  })
+}
+
+function edit(templates, admin_group, req, res) {
+  if (!req.user || !req.user.isAdmin) {
+    res.sendStatus(403)
+    return
+  }
+
+  var admin_group_id = req.params.admin_group_id
+
+  admin_group.getAdminGroups({ admin_group_id: admin_group_id }).then(([selected_admin_group]) => {
+    admin_group.getAdminGroupNames().then((admin_groups) => {
+      var html = templates.admin_group.edit({
+        user: req.user,
+        verb: 'Edit',
+        selected_admin_group: selected_admin_group,
+        admin_groups: admin_groups
+      })
+
+      res.send(html)
+    })
+  }).catch(err => {
+    console.error(err)
+    res.sendStatus(500)
+  })
+}
+
+function post(admin_group, req, res) {
```

Comment:
It appears as thought admins may be able to edit *any* group here, not just their own or lower.